temporal/build_id_versioning.go

```diff
@@ -0,0 +1,43 @@
+// The MIT License
+//
+// Copyright (c) 2022 Temporal Technologies Inc.  All rights reserved.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package temporal
+
+import "go.temporal.io/sdk/internal"
+
+// VersioningIntent indicates whether the user intends certain commands to be run on
+// a compatible worker build ID version or not.
+type VersioningIntent = internal.VersioningIntent
+
+const (
+	// VersioningIntentUnspecified indicates that the SDK should choose the most sensible default
+	// behavior for the type of command, accounting for whether the command will be run on the same
+	// task queue as the current worker.
+	VersioningIntentUnspecified = internal.VersioningIntentUnspecified
+	// VersioningIntentCompatible indicates that the command should run on a worker with compatible
+	// version if possible. It may not be possible if the target task queue does not also have
+	// knowledge of the current worker's build ID.
+	VersioningIntentCompatible = internal.VersioningIntentCompatible
+	// VersioningIntentUseDefault indicates that the command should run on the target task queue's
+	// current overall-default build ID.
+	VersioningIntentUseDefault = internal.VersioningIntentUseDefault
```

Comment:
Can we be consistent with the naming here?
Should be `UseCompatible` and `UseDefault` or drop the `Use` for both. I'd just drop `Use`.

Reply:
Sure